core/examples/hello_world.rs

```diff
@@ -17,7 +17,6 @@ use deno_core::*;
 fn op_sum(
   _state: &mut OpState,
   nums: Vec<f64>,
-  _: (),
```

Comment:
Aren't there more places where this can be removed?

Reply:
Rep but given that we'll also make OpState optional I wanted to do a cleanup PR focused on changing op arity.

In some cases we might want to flatten args and increase arity, in others we'll strip unused args

Reply:
If done now, it will make it clear this PR is not a net addition of code

Reply:
Did it the last commit, though it makes this PR a mix of feat & cleanup